### PR TITLE
fix point cloud bugs

### PR DESCRIPTION
**Pull request by sashankaryal**
## What changes are proposed in this pull request?

1. Add grid configurator
grid

2. Fix rgb shader not responding to point size change

3. Center points geometry

4. Use `points.clone()` to do a shallow copy so that point clouds with same URL are treated as different copies in the scene graph

## How is this patch tested? If it is not, please explain why.

Manual testing

## Summary by CodeRabbit

- **New Features**
  - Introduced `GridConfigurator` component for enhanced grid settings customization.
  - Added new grid settings with various configurable parameters.

- **Improvements**
  - Updated `PopoutSectionTitle` component styling for better alignment and spacing.
  - Converted `ActionPopOver` component to use `forwardRef` for improved ref forwarding.
  - Enhanced `Gizmos` component with dynamic grid settings.

- **Bug Fixes**
  - Replaced `useMemo` with `useEffect` in `Pcd` component for better point cloud data handling.

- **Performance**
  - Increased maximum value for `pointSize` in point cloud material controls from 20 to 50.

- **Chores**
  - Updated Docker build script to exclude unnecessary directories and build the `screenshot` Docker image.

### CODE REVIEW

File: app/packages/looker-3d/src/fo3d/Gizmos.tsx [hunk 1]
```diff
@@ -20,7 +27,7 @@ const FoAxesHelper = ({
 }) => {
   const { upVector } = useFo3dContext();
   const size = useMemo(
-    () => maxInOrthonormalPlane * 20,
+    () => maxInOrthonormalPlane * 10 + 100,
```
Comment (ritch) [inline app/packages/looker-3d/src/fo3d/Gizmos.tsx:30]: Could you make these easier to understand...? I have not idea why you are adding these random numbers here.

File: app/packages/looker-3d/src/fo3d/Gizmos.tsx [hunk 2]
```diff
@@ -99,34 +106,35 @@ export const Gizmos = () => {
...
             cellSize={cellSize}
             sectionSize={sectionSize}
             sectionColor={GRID_SECTION_COLOR}
             cellColor={GRID_CELL_COLOR}
             fadeDistance={fadeDistance}
-            fadeStrength={1}
-            followCamera
+            fadeStrength={shouldFade ? 1 : 0}
+            followCamera={shouldFade}
+            fadeFrom={0.5}
```
Comment (ritch) [inline app/packages/looker-3d/src/fo3d/Gizmos.tsx:137]: This is a lot more configurable than blender... lol

image

File: app/packages/looker-3d/src/fo3d/point-cloud/Pcd.tsx [hunk 3]
```diff
@@ -32,7 +32,19 @@ export const Pcd = ({
...
+
+  // todo: hack until https://github.com/pmndrs/react-three-fiber/issues/245 is fixed
+  const points = useMemo(() => points_.clone(false), [points_]);
+
+  useEffect(() => {
+    if (!points) {
+      return;
+    }
+
+    points.geometry.center();
```
Comment (ritch) [inline app/packages/looker-3d/src/fo3d/point-cloud/Pcd.tsx:45]:
```suggestion
 ```suggestion
      if (points) {
        points.geometry.center();
      }
```
?

File: app/packages/looker-3d/src/hooks/use-pcd-material-controls.ts [hunk 4]
```diff
@@ -34,7 +34,7 @@ export const usePcdMaterialControls = (
           pointSize: {
             value: pointSize ?? 1,
             min: 0.01,
-            max: 20,
+            max: 50,
```
Comment (ritch) [inline app/packages/looker-3d/src/hooks/use-pcd-material-controls.ts:37]: Arbitrary number? Add a comment?

File: app/packages/looker-3d/src/state.ts [hunk 5]
```diff
@@ -57,6 +57,44 @@ export const isGridOnAtom = atom<boolean>({
   ],
 });
 
+export const gridCellSizeAtom = atom<number>({
```
Comment (ritch) [inline app/packages/looker-3d/src/state.ts:60]: I wish recoil had a way to logically group state... lots of repetition here but just a side comment.

**Review by ritch — DISMISSED** [review]

File: app/packages/looker-3d/src/fo3d/Gizmos.tsx (hunk 2 above)
Comment (sashankaryal) [reply app/packages/looker-3d/src/fo3d/Gizmos.tsx:137]: Yeah... 😅 
We expose "fading", "infinite / non-infinite", "cell size", "section size", and "grid size" if non-infinite.

**Review by ritch — DISMISSED** [review]

**Review by ritch — APPROVED** [review]

**Pull request merged**